java-frontend/src/main/java/org/sonar/java/se/checks/CustomUnclosedResourcesCheck.java

```diff
@@ -119,7 +121,15 @@ private static String name(Tree tree) {
     if (tree.is(Tree.Kind.NEW_CLASS)) {
       return ((NewClassTree) tree).symbolType().name();
     }
-    return ((MethodInvocationTree) tree).symbolType().name();
+    MethodInvocationTree mit = (MethodInvocationTree) tree;
+    if (mit.symbolType().is("void")) {
+      Type type = mit.methodSelect().symbolType();
+      if (type.isUnknown()) {
+        return SyntaxTreeNameFinder.getName(mit.methodSelect());
```

Comment:
I removed the if after rewriting using `owner` as you suggested. No longer necessary to cover this.